src/Type/ArrayType.php

```diff
@@ -317,6 +317,18 @@ public function inferTemplateTypes(Type $receivedType): TemplateTypeMap
 		return $keyTypeMap->union($itemTypeMap);
 	}
 
+	public function map(callable $cb): Type
+	{
+		$keyType = $this->keyType->map($cb);
+		$itemType = $this->itemType->map($cb);
+
+		if ($keyType !== $this->keyType || $itemType !== $this->itemType) {
```

Comment:
Comparing objects with `!==`? Isn't it a mistake?

Reply:
It's on purpose: I want to check if `$keyType` is the same instance as `$this->keyType` here. Because if they do, I don't need to create a new `ArrayType` instance.